File: foliage-sample/src/main/java/org/lavenderx/foliage/sample/RpcBootstrap.java
package org.lavenderx.foliage.sample;

import org.lavenderx.foliage.sample.config.ServerConfig;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;
import org.springframework.context.support.AbstractApplicationContext;

public class RpcBootstrap {

    public static void main(String[] args) {
        final AbstractApplicationContext ctx =
                new AnnotationConfigApplicationContext(ServerConfig.class);

        Runtime.getRuntime().addShutdownHook(new Thread(ctx::close));
    }
}

File: foliage-sample/src/main/kotlin/org/lavenderx/foliage/sample/RpcBootstrap.kt
package org.lavenderx.foliage.sample

import org.lavenderx.foliage.sample.config.ServerConfig
import org.springframework.context.annotation.AnnotationConfigApplicationContext

class RpcBootstrap {
    companion object {
        @JvmStatic fun main(args: Array<String>) {
            val ctx = AnnotationConfigApplicationContext(ServerConfig::class.java)

            Runtime.getRuntime().addShutdownHook(Thread(Runnable { ctx.close() }))
        }
    }
}